joblib/test/test_memmapping.py

```diff
@@ -509,3 +510,16 @@ def test_pool_get_temp_dir(tmpdir):
     if sys.platform.startswith('win'):
         assert shared_mem is False
     assert pool_folder.endswith(pool_folder_name)
+
+
+def test_numpy_arrays_use_different_memory():
+    def func(arr, value):
+        arr[:] = value
+        return arr
+
+    arrays = [np.zeros((10, 10), dtype='float64') for i in range(10)]
+
+    results = Parallel(mmap_mode='w+', max_nbytes=0, n_jobs=2)(
+        delayed(func)(arr, i) for i, arr in enumerate(arrays))
+    rmax = [r.max() for r in results]
+    assert len(rmax) == len(set(rmax))
```

Comment:
I think the test looks good although I am not sure what you are trying to test with this assertion. Should you not check something like:
```py
for i, arr in enumerate(results):
    assert_array_equal(arr, i)
```

Reply:
The goal with this test was to check that all of the arrays are being mapped to different memory (coercing to set would remove duplicates). But it may be a bit obscure, and it's causing failures for some Python versions, so I'll change it.